First draft of exercises episode (#41)
carpentries/lesson-development-training: episodes/11-exercises.md

working memory into long-term memory.
One of the ways lesson developers can aid in this process is through exercises.
In addition to providing formative assessments for instructors, exercises help move 
new skills and concepts into long-term memory by providing learners an opportunity 
to practice what was recently learned.
Exercises should occur regularly throughout the lesson because they move items 
to long-term memory and free up learners' working memory for new items.

Creating exercises builds upon the learning objectives you created earlier in the lesson design process.
You can design exercises based on the actions/skills you described in your 
learning objectives (the learning outcomes you intend for the lesson).
You can benefit here from creating learning objectives with specific action verbs.
Specific verbs can help you decide what action you want the learners to perform in the exercise.
E.g. actions such as "explain" and "describe" may be better assessed by discussions 
and multiple choice questions, while "solve," "construct," "test" and other 
higher-level cognitive skills may be better assessed by debugging tasks, code-and-run, 

> or use-in-a-different-context exercises.

Reviewer: It is possible to link to specific sections of TTT, using URL anchors based on the `id` property of the section heading. For example, the URL for the Code & Run example is http://teachtogether.tech/en/index.html#code-run

However, I don't see a specific section for debugging tasks or use-in-a-different-context.

> or use-in-a-different-context exercises.

Author: Thanks Toby! For some reason I was trying to included nested headings in the urls.

::::::::::::::::::::::::::::::::::::::::::::::::::

As you discussed with your group in the last exercise, 
different types of learning objectives work better for novices,
while others are a better fit for competent practitioners or experts.
For example, novices benefit significantly from worked examples (fully solved example problems)
as it reduces their cognitive load 

> and worked examples are not as helpful in building skill for individuals with more expertise as they have to match up the information given with their mental model which can increase the cognitive load [Kirschner, 2006](https://github.com/carpentries/instructor-training/blob/gh-pages/files/papers/kirschner-minimal-guidance-fails-2006.pdf) <!--- Maybe this should reference Kalyuga et al 2001 and 2003 instead or in addition? Or the review  of expertise reversal effect - https://doi.org/10.1007/s10648-007-9054-3.  Also might have brought up worked example too soon as it is in the next section of notes -->

Author: I'd be okay with including Bloom's but I'm concerned about talking about it as hierarchical.  There is some criticism of using Bloom's as a hierarchy.  Some quick, not sure if they are the best references about it [1][1] [2][2].

[1]: https://www.edweek.org/education/opinion-heres-whats-wrong-with-blooms-taxonomy-a-deeper-learning-perspective/2018/03
[2]: https://tc2.ca/uploads/PDFs/Critical%20Discussions/unfortunate_consequences_blooms_taxonomy.pdf

> and worked examples are not as helpful in building skill for individuals with more expertise as they have to match up the information given with their mental model which can increase the cognitive load [Kirschner, 2006](https://github.com/carpentries/instructor-training/blob/gh-pages/files/papers/kirschner-minimal-guidance-fails-2006.pdf) <!--- Maybe this should reference Kalyuga et al 2001 and 2003 instead or in addition? Or the review  of expertise reversal effect - https://doi.org/10.1007/s10648-007-9054-3.  Also might have brought up worked example too soon as it is in the next section of notes -->

Reviewer: I have had a go at resolving this with #43, which is a PR to this branch.

> and worked examples are not as helpful in building skill for individuals with more expertise as they have to match up the information given with their mental model which can increase the cognitive load [Kirschner, 2006](https://github.com/carpentries/instructor-training/blob/gh-pages/files/papers/kirschner-minimal-guidance-fails-2006.pdf) <!--- Maybe this should reference Kalyuga et al 2001 and 2003 instead or in addition? Or the review  of expertise reversal effect - https://doi.org/10.1007/s10648-007-9054-3.  Also might have brought up worked example too soon as it is in the next section of notes -->

Author: Thanks!  Merged!

::::::::::::::::::::::::::::::::::::::::::::::::::

## Demo of Writing an Exercise 

Well-designed exercises are one of the most valuable resources for an instructor and 
any time spent on this is well invested.

<!--- will there be an example lesson that the trainer is working on ? -->

To create an exercise in the Carpentries Workbench, 
you can use colon-delimited sections called 'fenced divs'.

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Author: ```suggestion
In fact there are many types of boxes in the lesson infrastucture, called "callout blocks", 
```

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Author: ```suggestion
In fact there are many types of boxes in the lesson infrastucture, called "callout blocks", 
```

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Author: ```suggestion
In fact there are many types of boxes in the lesson infrastucture, called "callout blocks", 
```

Author: github won't let me commit this suggestion so fixing manually

with the class.
- **Sticky notes and written feedback** - Learners each get two colors of sticky notes. 
If they fall behind or run into an issue, they can put up the 
color that signals they need help.
If they have completed and exercise or think the instructor could speed up, they put their
other sticky note up.
Then at breaks these stickies can turn into written feedback "minute cards", where the learner 
writes down one thing they were confused about or that could be improved and one thing that went well
or they really liked learning.
- **Reflective assessment** - Learners spend time reflecting on what they have learned so far.
Reflection aids in transferring newly learned concepts into long-term memory and can be really helpful for "metacognition", where learners think about the process of learning, becoming aware of their progress towards acquiring new skills.
Developing metacognition improves learners' ability to guide their own learning on a topic after they have finished the lesson.
Reflection exercises can include using what was taught to label a diagram, draw a concept map, free write, or fill in a guided worksheet. 

::::::::::::::::::::::::::::::::::::::  challenge

> ## Exercise: More Practice with Callout Blocks (10 minutes)

Reviewer: ```suggestion
## Exercise: More Practice with Fenced Divs (10 minutes)
```

Return to the bulleted list of lesson objectives you added to the `index.md` file of your lesson and

> use callout blocks to display it in a formatted box with the `.prereq` class.

Reviewer: ```suggestion
use fenced divs to display it in a formatted box with the `.prereq` class.
```

::::::::::::::::::::::::::::::::::::::::::::::::::

## Demo of Writing an Exercise 

Well-designed exercises are one of the most valuable resources for an instructor and 
any time spent on this is well invested.

<!--- will there be an example lesson that the trainer is working on ? -->

To create an exercise in the Carpentries Workbench, 
you can use colon-delimited sections called 'fenced divs'.

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Author: I'm happy to fix this but still a little unclear on what I should call them.  I find the "Callouts" names a bit confusing though as these blocks can include challenges or exercises and not just callout blocks.  I'm happy to go with whichever we decide though.

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Reviewer 2: For what it's worth, the Quarto engine uses the term "callout" for these kind of blocks as well: https://quarto.org/docs/authoring/callouts.html

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Reviewer: I think we should stick to calling them "fenced divs". We will need to acknowledge that some fenced divs don't result in their own individual block (see `objectives` for example) but I think that will still be less confusing than referring to callouts.

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 

Author: Adding note in the last exercise that the objectives fenced divs all collapse into one box at the top of each episode.

> In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 
> that use fenced divs.

Reviewer: ```suggestion
In fact [there are many types of boxes in the lesson infrastucture that use fenced divs](https://carpentries.github.io/sandpaper-docs/episodes.html#callout-blocks).
```

@zkamvar is there a better place to link to than this e.g. a single place that lists all the valid classes of fenced div?

Reviewer 2: Oh hey! Yes, it is semi-documented [in {pegboard} with `validate_divs()`](https://carpentries.github.io/pegboard/articles/validation.html#fenced-div-validation), but I need to create the actual component guide 🙃

> To start a challenge fenced div the line must contain at least 3 colons, then the `challenge` tag.

Reviewer 2: At the moment, there's no plan to add "exercise" as a valid type, especially if it can refer to two valid types (that being said, having "exercise" be a generic type for "discussion" if there is a solution block and "challenge" if there is no solution block is an interesting concept to explore in the future)

> To start a challenge fenced div the line must contain at least 3 colons, then the `challenge` tag.

Reviewer 2: At the moment, [the fenced-div validator](https://carpentries.github.io/pegboard/articles/validation.html#fenced-div-validation) will catch those and throw the error `file:#xx  [unknown div] exercise`, but I think I can add extra context for that particular one to suggest "did you mean challenge or discussion?"

::::::::::::::::::::::::::::::::::::::  challenge

## Exercise: Formatting Exercises in a Lesson Site (15 minutes)

Using the approach demonstrated above,
format the MCQ you designed previously as an exercise in your lesson site.

::::::::::::::::::::::::::::::::::::::::::::::::::

Some other types of formative assessment include:

- **Think, pair, share** - Learners _think_ about an answer to a question, _pair_ up
with a classmate to discuss their answer, and then _share_ out the consensus they came to 
with the class.
- **Sticky notes and written feedback** - Learners each get two colors of sticky notes. 
If they fall behind or run into an issue, they can put up the 
color that signals they need help.
If they have completed and exercise or think the instructor could speed up, they put their
other sticky note up.
Then at breaks these stickies can turn into written feedback "minute cards", where the learner 
writes down one thing they were confused about or that could be improved and one thing that went well
or they really liked learning.
- **Reflective assessment** - Learners spend time reflecting on what they have learned so far.

> This can be really helpful for "metacognition", where learners think about the process of learning themselves.

Author: ```suggestion
Reflection aids in transferring newly learned concepts into long-term memory and can be really helpful for "metacognition", where learners think about the process of learning themselves.
Reflection exercises can include using what was taught to labelling a diagram, drawing a concept map, free write, fill in a guided workshop. 
```
@anenadic How do you feel about this change?

Author: Thanks so much for your feedback @anenadic!  Made some changes based on it, would appreciate another round of review.

I replied in line about my Bloom's hierarchy concerns. Upon reading your summary message I want to add that I think Bloom's is actually much better discussed in the Learning Objectives episode and then just referenced here.

Author: Thanks for that addition! Updated above.

::::::::::::::::::::::::::::::::::::::::::::::::::

## Demo of Writing an Exercise 

Well-designed exercises are one of the most valuable resources for an instructor and 
any time spent on this is well invested.

<!--- will there be an example lesson that the trainer is working on ? -->

> To create an exercise in the Carpentries lesson infrastructure, 

Reviewer 2: Optional: I'm trying to make "Carpentries Workbench" a thing, but "lesson infrastructure" is perfectly fine.

```suggestion
To create an exercise in the Carpentries Workbench, 
```

you can use colon-delimited sections called 'fenced divs'.
In fact there are many types of boxes in the lesson infrastucture, called "special blockquotes", 
that use fenced divs.
To start a challenge fenced div the line must contain at least 3 colons, then the `challenge` tag.
Then the content of the challenge is included on the following lines.
Finally, you can close the fenced div using another line with least 3 colons.

> ```

Reviewer 2: You can use the markdown tag and it will highlight markdown

````suggestion
```markdown
````

:::::::::::::::::::::::::::::::::::::: challenge

### Challenge Title

Challenge text, code, and other information goes here

::::::::::::::::::::::::::::::::::::::::::::::::::

```

If you also want to include an expandable solution box for the challenge you can 
add a solution fenced div within the challenge box.
The format is the same as for a challenge except the tag is `solution` instead. 
Note the solutions can all be expanded for more accessible reading using the "Expand All Solutions"
option at the top of each episode.

```
:::::::::::::::::::::::::::::::::::::: challenge

### Challenge Title

Challenge text, code, and other information goes here

:::::::::::::: solution

### Solution Title

Solution text, code, and other information

:::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::::::::::::::::

```

> For readability, you may want to try to match have final lines match up in length to
> the opening lines.

Reviewer 2: I want to avoid people assuming that the number of colons needs to match up exactly. 

```suggestion
For readability, you may want to try to match have final lines approximately match up 
in length to the opening lines.
```

> For readability, you may want to try to match have final lines approximately match up 
> in length to the opening lines.

Author: ```suggestion
For readability, you may want to have the length of the closing lines match the opening lines.
```

::::::::::::::::::::::::::::::::::::::  challenge

## Exercise: Formatting Exercises in a Lesson Site (15 minutes)

Using the approach demonstrated above,
format the MCQ you designed previously as an exercise in your lesson site.

::::::::::::::::::::::::::::::::::::::::::::::::::

Some other types of formative assessment include:

- **Think, pair, share** - Learners _think_ about an answer to a question, _pair_ up
with a classmate to discuss their answer, and then _share_ out the consensus they came to 
with the class.
- **Sticky notes and written feedback** - Learners each get two colors of sticky notes. 
If they fall behind or run into an issue, they can put up the 
color that signals they need help.
If they have completed and exercise or think the instructor could speed up, they put their
other sticky note up.
Then at breaks these stickies can turn into written feedback "minute cards", where the learner 
writes down one thing they were confused about or that could be improved and one thing that went well
or they really liked learning.
- **Reflective assessment** - Learners spend time reflecting on what they have learned so far.

> This can be really helpful for "metacognition", where learners think about the process of learning themselves.

Author: Think this is covered by the latest version.  Please let me know if there is more you would like to see @anenadic

:::::::::::::::::::::::::::::::::::::::: keypoints

- Exercises are important for learners to move what they've learned to long-term memory.
- Some types of exercises are better for particular audiences and to address certain objectives.

> - Exercises(and solutions) go in challenge callout blocks using fenced divs in the lesson.

Reviewer: ```suggestion
- Exercises (and solutions) go in blocks using fenced divs in the lesson.
```
